RocketChat/Rocket.Chat
# fix: Placeholder Issue with Microsoft Translate

## Proposed changes (including videos or screenshots)

## REASON BEHIND CHANGES
The issue lies in the` _translateMessage `function, which currently translates individual sentences while ignoring elements like '@username'. To address this, I modified the function to replace these elements with placeholders during translation. After the translation process, we can then replace these placeholders with the original usernames
## Issue(s)
This should close issues :-
fix #31146


## Steps to test or reproduce
1.Set up a Microsoft Azure account.
2.Create a global resource group for the translator.
3.Obtain an Ocp-Apim subscription key from the "Keys and Endpoint" section within the resource group.
4.Go to Rocket.Chat and navigate to "Workspace" => "Settings" => "Message".
5.Enable Auto-Translate.
6.Select the Service Provider "Microsoft".
7.Enter the Ocp-Apim key under "Microsoft" => "Ocp-Apim-Subscription-Key". 
8.Observe that parts of a chat message that are not supposed to be translated (like @ user xy) are shown as placeholders when using auto translate.


## Further comments
